062-quotas-management.md

```diff
@@ -0,0 +1,168 @@
+* The other configuration options for the built-in Kafka plugin can be configured by the user in the `.spec.kafka.config` section in the `Kafka` CR.
+
+The second supported plugin - the Strimzi Quotas plugin - will be configured as `type: strimzi`.
+The API will allow to configure the storage limits at which the plugin will start limiting the producers in order to prevent the disk from getting full.
+
+```yaml
+apiVersion: kafka.strimzi.io/v1beta2
+kind: Kafka
+metadata:
+  name: my-cluster
+spec:
+  kafka:
+    quotas:
+      type: strimzi
+      produceQuota: 1000000
+      fetchQuota: 1000000
+      minAvailableBytesPerVolume: 5368709120
+      excludedPrincipals:
+        - my-special-user
+    # ... 
+  # ...
+```
+
+The `type: strimzi` quotas API will support the following fields:
+* `produceQuota` for setting the default produce quota
+* `fetchQuota` for setting the default fetch quota
+* `minAvailableBytesPerVolume` for setting the minimal required free space per volume as number of bytes (this field can be configured only when `minAvailableRatioPerVolume`is not set)
+* `minAvailableRatioPerVolume` for setting the minimal required free space per volume as percentage of the overall volume size (this field can be configured only when `minAvailableBytesPerVolume`is not set)
+* `excludedPrincipals` for setting a list of users to be excluded from the applied quotas
+
+Other options might be configured directly by the user in `.spec.kafka.config` section of the `Kafka` CR.
+
+Based on this configuration, the operator will:
+* Set the `client.quota.callback.class` field in Kafka configuration to `io.strimzi.kafka.quotas.StaticQuotaCallback`.
+* Set the configuration on how to connect to the Kafka broker as a client using the `client.quota.callback.static.kafka.admin.` options.
+* Configures the default produce and fetch quota (`client.quota.callback.static.produce` and `client.quota.callback.static.fetch`).
+* Configure the minimal available disk space per-volume using the `client.quota.callback.static.storage.per.volume.limit.min.available.bytes` and `client.quota.callback.static.storage.per.volume.limit.min.available.ratio` options.
+* Set the excluded users for that the quotas will not be applied using the `client.quota.callback.static.excluded.principal.name.list` option. 
+  The list of excluded users will also automatically (regardless whether user specified some custom excluded users or not) include the internal users used by Topic Operator and Cruise Control to not block them as they might be useful to resolve the disk space issue:
+  * Topic Operator by changing the retention of the existing topics. 
+    (Producing / fetching messages is required only by the Bidirectional Topic Operator. So the TO user can be removed only BTO is not supported anymore.)
+  * Cruise Control by rebalancing the topics between the brokers to address the disk space shortage.
+
+The quota plugin configuration will not be set a forbidden configuration.
```

Comment:
```suggestion
It will not be possible to set a forbidden configuration for the Strimzi Quota plugin.
```

Reply:
I may have missed the point here. If so, maybe this _forbidden_ aspect needs some clarification.